### ISSUE TITLE
Anvil fork throws an error when receiving large gas limits 

### ISSUE THREAD

**Issue by Van0k** [foundry-rs/foundry#3573]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

anvil 0.1.0 (22ad954 2022-10-29T00:06:21.340549Z)

### Operating System

macOS (Intel)

### Describe the bug

Anvil throws HttpProviderError when a custom gas limit is passed as kwarg to an ethers contract function, that is larger than the block limit of the forked network.
For example: 
```
  const results = await this.pathFinder.callStatic.findAllSwaps(swapTask, {
    gasLimit: 400e6
  });
```
would fail when run on Anvil forking the mainnet. This is despite a larger block limit being set in `foundry.toml`.
This breaks functionality involving static calls that consume more than a block of gas - even though node providers allow up to 300-500M of gas per static call. `hardhat node` in forking mode also allows large static calls, as long as a custom gas limit is provided.

**Comment by mattsse** [comment]
what's the actual error message here?

**Comment by 0xmikko** [comment]
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"gasLimit":{"type":"BigNumber","hex":"0x11e1a300"},"to":"0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696","data":"0x252dba420000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000160000000000000000000000000000000000000000000000000000000000000022000000000000000000000000000000000000000000000000000000000000002e000000000000000000000000000000000000000000000000000000000000003a00000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000b0e2061b34171edc2b3c34e86b0b1f05ec7eeffc000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000614d98a57a5d879d717152de0690ed2b04562ade000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000cdc8ba9815c76e838a759e932f57bf6f990a6e71000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000bdfa4f4492dd7b7cf211209c4791af8d52bf5c50000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f70000000000000000000000000035fc5208ef989c28d47e552e92b0c507d2b31800000000000000000000000000000000000000000000000000000000","accessList":null}, error={"stack":"Error: processing response error (body=\"{\\\"jsonrpc\\\":\\\"2.0\\\",\\\"id\\\":79,\\\"error\\\":{\\\"code\\\":-32603,\\\"message\\\":\\\"EVM error CallerGasLimitMoreThenBlock\\\"}}\", error={\"stack\":\"Error: EVM error CallerGasLimitMoreThenBlock\\n    at getResult (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:142:28)\\n    at processJsonFunc (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:383:22)\\n    at /Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:320:42\\n    at step (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:33:23)\\n    at Object.next (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:14:53)\\n    at fulfilled (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:5:58)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"EVM error CallerGasLimitMoreThenBlock\",\"code\":-32603}, requestBody=\"{\\\"method\\\":\\\"eth_call\\\",\\\"params\\\":[{\\\"gas\\\":\\\"0x11e1a300\\\",\\\"to\\\":\\\"0x5ba1e12693dc8f9c48aad8770482f4739beed696\\\",\\\"data\\\":\\\"0x252dba420000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000160000000000000000000000000000000000000000000000000000000000000022000000000000000000000000000000000000000000000000000000000000002e000000000000000000000000000000000000000000000000000000000000003a00000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000b0e2061b34171edc2b3c34e86b0b1f05ec7eeffc000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000614d98a57a5d879d717152de0690ed2b04562ade000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000cdc8ba9815c76e838a759e932f57bf6f990a6e71000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000bdfa4f4492dd7b7cf211209c4791af8d52bf5c50000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f70000000000000000000000000035fc5208ef989c28d47e552e92b0c507d2b31800000000000000000000000000000000000000000000000000000000\\\"},\\\"0xf273f6\\\"],\\\"id\\\":79,\\\"jsonrpc\\\":\\\"2.0\\\"}\", requestMethod=\"POST\", url=\"http://localhost:8545\", code=SERVER_ERROR, version=web/5.7.1)\n    at Logger.makeError (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)\n    at Logger.throwError (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)\n    at /Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:341:28\n    at step (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:33:23)\n    at Object.next (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:14:53)\n    at fulfilled (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:5:58)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","message":"processing response error (body=\"{\\\"jsonrpc\\\":\\\"2.0\\\",\\\"id\\\":79,\\\"error\\\":{\\\"code\\\":-32603,\\\"message\\\":\\\"EVM error CallerGasLimitMoreThenBlock\\\"}}\", error={\"stack\":\"Error: EVM error CallerGasLimitMoreThenBlock\\n    at getResult (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:142:28)\\n    at processJsonFunc (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:383:22)\\n    at /Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:320:42\\n    at step (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:33:23)\\n    at Object.next (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:14:53)\\n    at fulfilled (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:5:58)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"EVM error CallerGasLimitMoreThenBlock\",\"code\":-32603}, requestBody=\"{\\\"method\\\":\\\"eth_call\\\",\\\"params\\\":[{\\\"gas\\\":\\\"0x11e1a300\\\",\\\"to\\\":\\\"0x5ba1e12693dc8f9c48aad8770482f4739beed696\\\",\\\"data\\\":\\\"0x252dba420000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000160000000000000000000000000000000000000000000000000000000000000022000000000000000000000000000000000000000000000000000000000000002e000000000000000000000000000000000000000000000000000000000000003a00000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000b0e2061b34171edc2b3c34e86b0b1f05ec7eeffc000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000614d98a57a5d879d717152de0690ed2b04562ade000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000cdc8ba9815c76e838a759e932f57bf6f990a6e71000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000bdfa4f4492dd7b7cf211209c4791af8d52bf5c50000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f70000000000000000000000000035fc5208ef989c28d47e552e92b0c507d2b31800000000000000000000000000000000000000000000000000000000\\\"},\\\"0xf273f6\\\"],\\\"id\\\":79,\\\"jsonrpc\\\":\\\"2.0\\\"}\", requestMethod=\"POST\", url=\"http://localhost:8545\", code=SERVER_ERROR, version=web/5.7.1)","reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":79,\"error\":{\"code\":-32603,\"message\":\"EVM error CallerGasLimitMoreThenBlock\"}}","error":{"stack":"Error: EVM error CallerGasLimitMoreThenBlock\n    at getResult (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:142:28)\n    at processJsonFunc (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:383:22)\n    at /Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/src.ts/index.ts:320:42\n    at step (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:33:23)\n    at Object.next (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:14:53)\n    at fulfilled (/Users/mikael/Coding/gearbox/liquidator-v2/node_modules/@ethersproject/web/lib/index.js:5:58)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","message":"EVM error CallerGasLimitMoreThenBlock","code":-32603},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"gas\":\"0x11e1a300\",\"to\":\"0x5ba1e12693dc8f9c48aad8770482f4739beed696\",\"data\":\"0x252dba420000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000160000000000000000000000000000000000000000000000000000000000000022000000000000000000000000000000000000000000000000000000000000002e000000000000000000000000000000000000000000000000000000000000003a00000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000b0e2061b34171edc2b3c34e86b0b1f05ec7eeffc000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000614d98a57a5d879d717152de0690ed2b04562ade000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000cdc8ba9815c76e838a759e932f57bf6f990a6e71000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f7000000000000000000000000bdfa4f4492dd7b7cf211209c4791af8d52bf5c50000000000000000000000000000000000000000000000000000000000000000000000000000000000a2ca503153cd5cb2892a0928ac0f71f49a3c194000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000440dbd616d000000000000000000000000672461bfc20dd783444a830ad4c38b345ab6e2f70000000000000000000000000035fc5208ef989c28d47e552e92b0c507d2b31800000000000000000000000000000000000000000000000000000000\"},\"0xf273f6\"],\"id\":79,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"http://localhost:8545"}, code=CALL_EXCEPTION, version=providers/5.7.2)

**Comment by Evalir** [comment]
Closing, fixed by #4324 and a few other gas estimation ports we got from Reth.
